### ISSUE TITLE
Weather Dependent Flight Planning

### ISSUE THREAD

**Issue by SnappyComebacks** [dcs-liberation/dcs_liberation#2641]
**Is your feature request related to a problem? Please describe.**

If AI is assigned a DEAD task in overcast weather, it will not complete the task.  Setting the weather to clear allows them to complete the task.  However, if the altitude of the flight is below the cloud base of overcast weather at the time when `Attack Group` starts, the flight is able to perform its attack.

**Describe the solution you'd like**

- Planner chooses not to plan ground attack missions (except SEAD and SEAD Escort?) when overcast, and getting low (setting waypoint altitudes below the cloud base) is too dangerous.  If it is safe (none, or mitigated SAM/AAA threats), fly under the cloud base.
- Manually planned ground attack missions in overcast weather default to under the cloud base.

**Additional context**

[AI_Ground_Attack_Weather_Dependent.zip](https://github.com/dcs-liberation/dcs_liberation/files/10249468/AI_Ground_Attack_Weather_Dependent.zip)

Above is a .miz (requires Persian Gulf map, if you raise the waypoint altitudes to 20k, the planes do not attack, if you keep them at 5k, they continue the attack.  Something strange is `Overcast 7` and `Overcast and Rain 1`, they won't attack from high altitude, but `Overcast and Rain 3`, they will.  Not sure if this is a DCS feature or bug.

**Comment by Starfire13** [comment]
Yeah, the AI won't do DEAD, BAI, and I think anti-convoy too in bad weather. Not sure about OCA/aircraft. I've just been manually changing the weather in the mission editor on every turn where I get overcast/rain, to avoid this. It's definitely an odd thing since supposedly the AI can see aircraft straight through cloud.

**Comment by SnappyComebacks** [comment]
> Yeah, the AI won't do DEAD, BAI, and I think anti-convoy too in bad weather. Not sure about OCA/aircraft.

Watching the Tacview, they didn't do CAS as well.

**Comment by DanAlbert** [comment]
Preventing planning for certain missions when the weather will interfere SGTM. I figured we were going to do that sooner or later.

The even better solution won't be an option until the flight plan profiles FR is done (that's what I'm planning to do after my current work, fwiw), but that will allow us to do things like "if it's overcast, plan low-level missions".
